src/main/scala/uk/gov/nationalarchives/FileProcessor.scala

```diff
@@ -57,28 +57,48 @@ class FileProcessor(
     } yield parsedJson
   }
 
+  def parseUri(potentialUri: Option[String]): IO[Option[ParsedUri]] = {
+    potentialUri.map { uri =>
+      val citeRegex = "^.*/id/([a-z]*)/".r
+      val uriWithoutDocTypeRegex = "(^.*/\\d{4}/\\d*)".r
```

Comment:
The spec we've had from TRE says that it will be year followed by a number at least. We have to work to that, there's no way we can account for every possible url combination so we'll have to trust that they send what is agreed.